builtin/credential/approle/backend.go

```diff
@@ -59,8 +60,8 @@ func Backend(conf *logical.BackendConfig) (*backend, error) {
 		// Set the salt object for the backend
 		salt: salt,
 
-		// Create the lock for making changes to the Roles registered with the backend
-		roleLock: &sync.RWMutex{},
+		// Create the map of locks to modify the registered roles
+		roleLocksMap: map[string]*sync.RWMutex{},
```

Comment:
As a minor optimization, since you know how many locks will go in there, you could use `make` and pass it 256 as the size.